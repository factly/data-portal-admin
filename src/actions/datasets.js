import axios from '../utils/axios';
import {
  ADD_DATASET,
  ADD_DATASET_FORMAT,
  REMOVE_DATASET_FORMAT,
  ADD_DATASETS,
  SET_DATASET_LOADING,
  SET_DATASET_REQUEST,
  SET_DATASET_IDS,
  RESET_DATASET,
  DATASET_API,
} from '../constants/datasets';
import { addCurrencies } from './currencies';
import { addTags } from './tags';
import { addMedium } from './media';
import { getIds, buildObjectOfItems, getValues, deleteKeys } from '../utils/objects';

export const loadDatasets = (page = 1, limit = 5) => {
  return async (dispatch, getState) => {
    const {
      datasets: { req },
    } = getState();

    let ids;
    for (let item of req) {
      if (item.page === page && item.limit === limit) {
        ids = [...item.ids];
      }
    }

    if (ids) {
      dispatch(setDatasetIds(ids));
      return;
    }

    dispatch(setLoading(true));

    const response = await axios({
      url: `${DATASET_API}?page=${page}&limit=${limit}`,
      method: 'get',
    });

    const { nodes, total } = response.data;
    const currentPageIds = getIds(nodes);
    const currentReq = { page: page, limit: limit, ids: currentPageIds };
    dispatch(setDatasetRequest(currentReq, total));
    dispatch(addDatasets(nodes));
    dispatch(setDatasetIds(currentPageIds));

    dispatch(setLoading(false));
  };
};

export const createDataset = (data) => {
  return async (dispatch) => {
    dispatch(setLoading(true));

    const response = await axios({
      url: DATASET_API,
      method: 'post',
      data: data,
    });

    dispatch(addDataset(response.data));
    dispatch(setLoading(false));

    return response.data;
  };
};

export const createDatasetFormat = (datasetId, data) => {
  return async (dispatch) => {
    dispatch(setLoading(true));

    const response = await axios({
      url: `${DATASET_API}/${datasetId}/format`,
      method: 'post',
      data: data,
    });

    dispatch(addDatasetFormat(datasetId, response.data));
    dispatch(setLoading(false));
  };
};

export const updateDataset = (id, data) => {
  return async (dispatch) => {
    let url = `${DATASET_API}/${id}`;

    const response = await axios({
      url: url,
      method: 'put',
      data: data,
    });

    dispatch(addDataset(response.data));

    return response.data;
  };
};

export const deleteDataset = (id) => {
  return async (dispatch, getState) => {
    let url = `${DATASET_API}/${id}`;

    dispatch(setLoading(true));

    await axios({
      url: url,
      method: 'delete',
    });

    dispatch(resetDataset());
    dispatch(setLoading(false));
  };
};

export const deleteDatasetFormat = (datasetId, id) => {
  return async (dispatch) => {
    let url = `${DATASET_API}/${datasetId}/format/${id}`;

    dispatch(setLoading(true));

    await axios({
      url: url,
      method: 'delete',
    });

    dispatch(removeDatasetFormat(datasetId, id));
    dispatch(setLoading(false));
  };
};

export const getDataset = (id) => {
  return async (dispatch, getState) => {
    const {
      datasets: { items },
    } = getState();

    if (items[id]) {
      return;
    }

    dispatch(setLoading(true));

    const response = await axios({
      url: `${DATASET_API}/${id}`,
      method: 'get',
    });
    dispatch(addDataset(response.data));

    dispatch(setLoading(false));
  };
};

export const setLoading = (loading) => {
  return {
    type: SET_DATASET_LOADING,
    payload: { loading },
  };
};

export const addDataset = (dataset) => (dispatch) => {
  const currencies = getValues([dataset], 'currency');
  dispatch(addCurrencies(currencies));

  const medium = getValues([dataset], 'featured_medium');
  dispatch(addMedium(medium));

  const tags = getValues([dataset], 'tags');
  dispatch(addTags(tags));

  dataset.tags = getIds(dataset.tags);

  dispatch({
    type: ADD_DATASET,
    payload: { dataset: deleteKeys([dataset], ['currency', 'featured_medium'])[0] },
  });
};

export const addDatasets = (datasets) => (dispatch) => {
  const currencies = getValues(datasets, 'currency');
  dispatch(addCurrencies(currencies));

  const medium = getValues(datasets, 'featured_medium');
  dispatch(addMedium(medium));

  const tags = getValues(datasets, 'tags');
  dispatch(addTags(tags));

  datasets.forEach((dataset) => {
    dataset.tags = getIds(dataset.tags);
  });

  dispatch({
    type: ADD_DATASETS,
    payload: {
      datasets: buildObjectOfItems(deleteKeys(datasets, ['currency', 'featured_medium'])),
    },
  });
};

export const addDatasetFormat = (datasetId, datasetFormat) => {
  return {
    type: ADD_DATASET_FORMAT,
    payload: { datasetId, datasetFormat },
  };
};

export const removeDatasetFormat = (datasetId, datasetFormatId) => {
  return {
    type: REMOVE_DATASET_FORMAT,
    payload: { datasetId, datasetFormatId },
  };
};

export const setDatasetRequest = (req, total) => {
  return {
    type: SET_DATASET_REQUEST,
    payload: { req, total },
  };
};

export const setDatasetIds = (ids) => {
  return {
    type: SET_DATASET_IDS,
    payload: { ids },
  };
};

export const resetDataset = () => {
  return {
    type: RESET_DATASET,
  };
};
