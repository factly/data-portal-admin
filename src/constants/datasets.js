export const LOADING_DATASETS = 'LOADING_DATASETS';
export const LOAD_DATASETS_SUCCESS = 'LOAD_DATASETS_SUCCESS';
export const SET_DATASETS_LIST_TOTAL = 'SET_DATASETS_LIST_TOTAL';
export const ADD_DATASETS_LIST_REQUEST = 'ADD_DATASETS_LIST_REQUEST';
export const SET_DATASETS_LIST_CURRENT_PAGE = 'SET_DATASETS_LIST_CURRENT_PAGE';
export const LOAD_DATASETS_FAILURE = 'LOAD_DATASETS_FAILURE';
export const CREATING_DATASET = 'CREATING_DATASET';
export const CREATE_DATASET_SUCCESS = 'CREATE_DATASET_SUCCESS';
export const CREATE_DATASET_FAILURE = 'CREATE_DATASET_FAILURE';
export const UPDATING_DATASET = 'UPDATING_DATASET';
export const UPDATE_DATASET_SUCCESS = 'UPDATE_DATASET_SUCCESS';
export const UPDATE_DATASET_FAILURE = 'UPDATE_DATASET_FAILURE';
export const DELETING_DATASET = 'DELETING_DATASET';
export const DELETE_DATASET_SUCCESS = 'DELETE_DATASET_SUCCESS';
export const DELETE_DATASET_FAILURE = 'DELETE_DATASET_FAILURE';

export const baseUrl = `${process.env.REACT_APP_API_URL}/datasets`;
