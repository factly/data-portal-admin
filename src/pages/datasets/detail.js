import React from 'react';
import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';

import DatasetDetails from './components/DatasetDetails';
import DatasetFormatsList from './components/DatasetFormatsList';

const DatasetDetail = () => {
  const { id } = useParams();
  const [datasetSampleId, setDatasetSampleId] = React.useState(null);
  const { dataset, datasetFormats } = useSelector(({ datasets }) => ({
    dataset: datasets.items[id],
    datasetFormats: datasets.items[id]?.formats,
  }));

  React.useEffect(() => {
    if (dataset?.sample_url) {
      const sample = datasetFormats.find((item) => item.url === dataset.sample_url);
      sample && setDatasetSampleId(sample.id);
    }
  }, [datasetFormats]);

  return (
    <>
      <DatasetDetails dataset={dataset} />
      <DatasetFormatsList
        datasetId={dataset.id}
        showOperations={false}
        datasetSampleId={datasetSampleId}
      />
    </>
  );
};

export default DatasetDetail;
