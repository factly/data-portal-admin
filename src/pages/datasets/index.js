import React from 'react';
import DatasetsList from './components/DatasetsList';
import { Button } from 'antd';
import { Link } from 'react-router-dom';

const Datasets = () => {
  return (
    <>
      <Link to={'/datasets/create'}>
        <Button type="primary" style={{ marginBottom: 16 }}>
          Add Dataset
        </Button>
      </Link>
      <DatasetsList />
    </>
  );
};

export default Datasets;
