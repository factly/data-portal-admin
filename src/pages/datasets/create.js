import React from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';

import DatasetCreateForm from './components/DatasetCreateForm';
import { createDataset, createDatasetFormat } from '../../actions/datasets';

const DatasetCreate = () => {
  const history = useHistory();
  const dispatch = useDispatch();

  const onCreate = async (values) => {
    return dispatch(createDataset(values));
    // history.push('/datasets');
  };

  const onCreateDatasetFormat = (datasetId, data) => {
    return dispatch(createDatasetFormat(datasetId, data));
  };

  return (
    <DatasetCreateForm onSubmitDataset={onCreate} onSubmitDatasetFormat={onCreateDatasetFormat} />
  );
};

export default DatasetCreate;
