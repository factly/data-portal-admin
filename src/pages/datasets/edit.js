import React from 'react';
import { useParams, useHistory } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { Skeleton } from 'antd';

import DatasetCreateForm from './components/DatasetCreateForm';
import { getDataset, updateDataset, createDatasetFormat } from '../../actions/datasets';

const EditDataset = () => {
  const { id } = useParams();
  const history = useHistory();
  const dispatch = useDispatch();
  const { dataset } = useSelector(({ datasets }) => datasets);

  React.useEffect(() => {
    dispatch(getDataset(id));
  }, []);

  if (!dataset.id) return <Skeleton />;

  const onUpdateDataset = (values) => {
    return dispatch(updateDataset(id, values));
    // history.push('/datasets');
  };

  const onCreateDatasetFormat = (datasetId, data) => {
    return dispatch(createDatasetFormat(datasetId, data));
  };

  if (typeof dataset.frequency === 'string') {
    dataset.frequency = {
      count: dataset.frequency?.split(' ')[0],
      units: dataset.frequency?.split(' ')[1],
    };
  }

  return (
    <DatasetCreateForm
      dataset={dataset}
      datasetFormats={dataset.formats}
      onSubmitDataset={onUpdateDataset}
      onSubmitDatasetFormat={onCreateDatasetFormat}
    />
  );
};

export default EditDataset;
