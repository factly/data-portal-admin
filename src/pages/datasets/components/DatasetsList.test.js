import React from 'react';
import { BrowserRouter, Router, useHistory } from 'react-router-dom';
import renderer, { act as rendererAct } from 'react-test-renderer';
import { useDispatch, Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { Popconfirm, Button, List, Card } from 'antd';
import { act } from '@testing-library/react';

import '../../../matchMedia.mock';
import DatasetList from './DatasetsList';
import { loadDatasets, deleteDataset } from '../../../actions/datasets';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

jest.mock('../../../actions/datasets', () => ({
  loadDatasets: jest.fn(),
  deleteDataset: jest.fn(),
}));

describe('DatasetsList component', () => {
  let store;
  let mockedDispatch;
  const dataset = {
    id: 1,
    title: 'title',
    featured_media: { id: 1, alt_text: 'alt_text', url: 'url' },
    contact_email: 'contact_email',
    contact_name: 'contact_name',
    license: 'license',
    source: 'source',
  };

  describe('snapshot testing', () => {
    beforeEach(() => {
      store = mockStore({
        datasets: {
          loading: false,
          ids: [1, 2],
          req: [
            {
              page: 1,
              limit: 5,
              ids: [1, 2],
            },
          ],
          items: {
            1: {
              id: 1,
              title: 'title',
              contact_email: 'contact_email',
              contact_name: 'contact_name',
              data_standard: 'data_standard',
              description: 'description',
              frequency: 'frequency',
              granularity: 'granularity',
              license: 'license',
              related_articles: 'related_articles',
              source: 'source',
              temporal_coverage: 'temporal_coverage',
              time_saved: 'time_saved',
              created_at: '2020-12-12',
            },
            2: {
              id: 2,
              title: 'title',
              contact_email: 'contact_email',
              contact_name: 'contact_name',
              data_standard: 'data_standard',
              description: 'description',
              frequency: 'frequency',
              granularity: 'granularity',
              license: 'license',
              related_articles: 'related_articles',
              source: 'source',
              temporal_coverage: 'temporal_coverage',
              time_saved: 'time_saved',
              created_at: '2020-12-12',
            },
          },
          total: 2,
        },
      });
      store.dispatch = jest.fn(() => ({}));
      mockedDispatch = jest.fn(() => Promise.resolve({}));
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should render the component', () => {
      const tree = renderer
        .create(
          <Provider store={store}>
            <BrowserRouter>
              <DatasetList />
            </BrowserRouter>
          </Provider>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should match component with no data', () => {
      store = mockStore({
        datasets: {
          loading: false,
          ids: [],
          req: [],
          items: {},
          total: 0,
        },
      });
      const tree = renderer
        .create(
          <Provider store={store}>
            <DatasetList />
          </Provider>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should match component with datasets', () => {
      let component;
      rendererAct(() => {
        component = renderer.create(
          <Provider store={store}>
            <BrowserRouter>
              <DatasetList />
            </BrowserRouter>
          </Provider>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      expect(mockedDispatch).toHaveBeenCalledTimes(1);
      expect(loadDatasets).toHaveBeenCalledWith(1, 10);
    });
  });
  describe('component testing', () => {
    let wrapper;
    beforeEach(() => {
      jest.clearAllMocks();
      store = mockStore({
        datasets: {
          loading: false,
          ids: [1, 2],
          req: [
            {
              page: 1,
              limit: 5,
              ids: [1, 2],
            },
          ],
          items: {
            1: {
              id: 1,
              title: 'title',
              contact_email: 'contact_email',
              contact_name: 'contact_name',
              data_standard: 'data_standard',
              description: 'description',
              frequency: 'frequency',
              granularity: 'granularity',
              license: 'license',
              related_articles: 'related_articles',
              source: 'source',
              temporal_coverage: 'temporal_coverage',
              time_saved: 'time_saved',
              created_at: '2020-12-12',
            },
            2: {
              id: 2,
              title: 'title',
              contact_email: 'contact_email',
              contact_name: 'contact_name',
              data_standard: 'data_standard',
              description: 'description',
              frequency: 'frequency',
              granularity: 'granularity',
              license: 'license',
              related_articles: 'related_articles',
              source: 'source',
              temporal_coverage: 'temporal_coverage',
              time_saved: 'time_saved',
              created_at: '2020-12-12',
            },
          },
          total: 2,
        },
      });
      store.dispatch = jest.fn(() => ({}));
      mockedDispatch = jest.fn(() => Promise.resolve({}));
      useDispatch.mockReturnValue(mockedDispatch);
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('should have empty', () => {
      store = mockStore({
        datasets: {
          loading: false,
          ids: [],
          req: [],
          items: {},
          total: 0,
        },
      });
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <BrowserRouter>
              <DatasetList />
            </BrowserRouter>
          </Provider>,
        );
      });
      const list = wrapper.find(List.Item);
      expect(list.length).toEqual(0);
    });
    it('should change the page', () => {
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <BrowserRouter>
              <DatasetList />
            </BrowserRouter>
          </Provider>,
        );
      });
      const list = wrapper.find(List);
      list.props().pagination.onChange(2);
      wrapper.update();
      const updatedList = wrapper.find(List);
      expect(updatedList.props().pagination.current).toEqual(2);
    });
    it('should edit dataset', () => {
      const historyMock = {
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn(),
      };
      wrapper = mount(
        <Provider store={store}>
          <Router history={historyMock}>
            <DatasetList />
          </Router>
          ,
        </Provider>,
      );

      const card = wrapper.find(Card).at(0);
      const button = card.find('li').at(0).find('span').at(1);
      button.simulate('click');
      expect(historyMock.push).toHaveBeenCalledWith('/datasets/1/edit');
    });
    it('should delete dataset', () => {
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <BrowserRouter>
              <DatasetList />
            </BrowserRouter>
          </Provider>,
        );
      });
      const card = wrapper.find(Card).at(0);
      const button = card.find('li').at(1).find('span').at(1);
      button.simulate('click');

      const popconfirm = wrapper.find(Popconfirm);
      popconfirm
        .findWhere((item) => item.type() === 'button' && item.text() === 'OK')
        .simulate('click');
      expect(deleteDataset).toHaveBeenCalled();
      expect(deleteDataset).toHaveBeenCalledWith(1);
      expect(loadDatasets).toHaveBeenCalledWith(1, 10);
    });
    it('should have no delete and edit buttons', () => {
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <BrowserRouter>
              <DatasetList />
            </BrowserRouter>
          </Provider>,
        );
      });

      const button = wrapper.find(Button);
      expect(button.length).toEqual(0);
    });
  });
});
