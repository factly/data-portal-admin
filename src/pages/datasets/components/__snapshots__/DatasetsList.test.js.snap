// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetsList component snapshot testing should match component with datasets 1`] = `
<div
  className="ant-list ant-list-split ant-list-grid ant-list-something-after-last-item"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -8,
            "marginRight": -8,
          }
        }
      >
        <div
          className="ant-col ant-col-6"
          style={
            Object {
              "paddingLeft": 8,
              "paddingRight": 8,
            }
          }
        >
          <div
            className="ant-list-item"
          >
            <div
              className="ant-card ant-card-hoverable"
            >
              <div
                className="ant-card-head"
                style={Object {}}
              >
                <div
                  className="ant-card-head-wrapper"
                >
                  <div
                    className="ant-card-head-title"
                  >
                    <a
                      href="/datasets/1"
                      onClick={[Function]}
                    >
                      title
                    </a>
                  </div>
                </div>
              </div>
              <div
                className="ant-card-cover"
              >
                <img
                  alt="alt_text"
                  className="photo"
                  src="url"
                />
              </div>
              <div
                className="ant-card-body"
                style={Object {}}
              >
                <div
                  className="ant-descriptions"
                >
                  <div
                    className="ant-descriptions-view"
                  >
                    <table>
                      <tbody>
                        <tr
                          className="ant-descriptions-row"
                        >
                          <td
                            className="ant-descriptions-item"
                            colSpan={1}
                          >
                            <span
                              className="ant-descriptions-item-label ant-descriptions-item-colon"
                            >
                              Contact email
                            </span>
                            <span
                              className="ant-descriptions-item-content"
                            >
                              contact_email
                            </span>
                          </td>
                        </tr>
                        <tr
                          className="ant-descriptions-row"
                        >
                          <td
                            className="ant-descriptions-item"
                            colSpan={1}
                          >
                            <span
                              className="ant-descriptions-item-label ant-descriptions-item-colon"
                            >
                              Contact name
                            </span>
                            <span
                              className="ant-descriptions-item-content"
                            >
                              contact_name
                            </span>
                          </td>
                        </tr>
                        <tr
                          className="ant-descriptions-row"
                        >
                          <td
                            className="ant-descriptions-item"
                            colSpan={1}
                          >
                            <span
                              className="ant-descriptions-item-label ant-descriptions-item-colon"
                            >
                              License
                            </span>
                            <span
                              className="ant-descriptions-item-content"
                            >
                              license
                            </span>
                          </td>
                        </tr>
                        <tr
                          className="ant-descriptions-row"
                        >
                          <td
                            className="ant-descriptions-item"
                            colSpan={1}
                          >
                            <span
                              className="ant-descriptions-item-label ant-descriptions-item-colon"
                            >
                              Source
                            </span>
                            <span
                              className="ant-descriptions-item-content"
                            >
                              source
                            </span>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
              <ul
                className="ant-card-actions"
              >
                <li
                  style={
                    Object {
                      "width": "50%",
                    }
                  }
                >
                  <span>
                    <span
                      aria-label="edit"
                      className="anticon anticon-edit"
                      onClick={[Function]}
                      role="img"
                      tabIndex={-1}
                    >
                      <svg
                        aria-hidden="true"
                        className=""
                        data-icon="edit"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                        />
                      </svg>
                    </span>
                  </span>
                </li>
                <li
                  style={
                    Object {
                      "width": "50%",
                    }
                  }
                >
                  <span>
                    <span
                      aria-label="delete"
                      className="anticon anticon-delete"
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                      role="img"
                      tabIndex={-1}
                    >
                      <svg
                        aria-hidden="true"
                        className=""
                        data-icon="delete"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                        />
                      </svg>
                    </span>
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-list-pagination"
  >
    <ul
      className="ant-pagination"
      style={Object {}}
      unselectable="unselectable"
    >
      <li
        aria-disabled={true}
        className="ant-pagination-prev ant-pagination-disabled"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={null}
        title="Previous Page"
      >
        <a
          className="ant-pagination-item-link"
          disabled={true}
        >
          <span
            aria-label="left"
            className="anticon anticon-left"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </span>
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={1}
      >
        <a>
          1
        </a>
      </li>
      <li
        aria-disabled={true}
        className="ant-pagination-next ant-pagination-disabled"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={null}
        title="Next Page"
      >
        <a
          className="ant-pagination-item-link"
          disabled={true}
        >
          <span
            aria-label="right"
            className="anticon anticon-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`DatasetsList component snapshot testing should match component with no data 1`] = `
<div
  className="ant-card ant-card-bordered"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-empty"
    >
      <div
        className="ant-empty-image"
      >
        <svg
          className="ant-empty-img-default"
          height="152"
          viewBox="0 0 184 152"
          width="184"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <g
              transform="translate(24 31.67)"
            >
              <ellipse
                className="ant-empty-img-default-ellipse"
                cx="67.797"
                cy="106.89"
                rx="67.797"
                ry="12.668"
              />
              <path
                className="ant-empty-img-default-path-1"
                d="M122.034 69.674L98.109 40.229c-1.148-1.386-2.826-2.225-4.593-2.225h-51.44c-1.766 0-3.444.839-4.592 2.225L13.56 69.674v15.383h108.475V69.674z"
              />
              <path
                className="ant-empty-img-default-path-2"
                d="M101.537 86.214L80.63 61.102c-1.001-1.207-2.507-1.867-4.048-1.867H31.724c-1.54 0-3.047.66-4.048 1.867L6.769 86.214v13.792h94.768V86.214z"
                transform="translate(13.56)"
              />
              <path
                className="ant-empty-img-default-path-3"
                d="M33.83 0h67.933a4 4 0 0 1 4 4v93.344a4 4 0 0 1-4 4H33.83a4 4 0 0 1-4-4V4a4 4 0 0 1 4-4z"
              />
              <path
                className="ant-empty-img-default-path-4"
                d="M42.678 9.953h50.237a2 2 0 0 1 2 2V36.91a2 2 0 0 1-2 2H42.678a2 2 0 0 1-2-2V11.953a2 2 0 0 1 2-2zM42.94 49.767h49.713a2.262 2.262 0 1 1 0 4.524H42.94a2.262 2.262 0 0 1 0-4.524zM42.94 61.53h49.713a2.262 2.262 0 1 1 0 4.525H42.94a2.262 2.262 0 0 1 0-4.525zM121.813 105.032c-.775 3.071-3.497 5.36-6.735 5.36H20.515c-3.238 0-5.96-2.29-6.734-5.36a7.309 7.309 0 0 1-.222-1.79V69.675h26.318c2.907 0 5.25 2.448 5.25 5.42v.04c0 2.971 2.37 5.37 5.277 5.37h34.785c2.907 0 5.277-2.421 5.277-5.393V75.1c0-2.972 2.343-5.426 5.25-5.426h26.318v33.569c0 .617-.077 1.216-.221 1.789z"
              />
            </g>
            <path
              className="ant-empty-img-default-path-5"
              d="M149.121 33.292l-6.83 2.65a1 1 0 0 1-1.317-1.23l1.937-6.207c-2.589-2.944-4.109-6.534-4.109-10.408C138.802 8.102 148.92 0 161.402 0 173.881 0 184 8.102 184 18.097c0 9.995-10.118 18.097-22.599 18.097-4.528 0-8.744-1.066-12.28-2.902z"
            />
            <g
              className="ant-empty-img-default-g"
              transform="translate(149.65 15.383)"
            >
              <ellipse
                cx="20.654"
                cy="3.167"
                rx="2.849"
                ry="2.815"
              />
              <path
                d="M5.698 5.63H0L2.898.704zM9.259.704h4.985V5.63H9.259z"
              />
            </g>
          </g>
        </svg>
      </div>
      <p
        className="ant-empty-description"
      >
        No Data
      </p>
    </div>
  </div>
</div>
`;

exports[`DatasetsList component snapshot testing should render the component 1`] = `
<div
  className="ant-card ant-card-bordered"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-empty"
    >
      <div
        className="ant-empty-image"
      >
        <svg
          className="ant-empty-img-default"
          height="152"
          viewBox="0 0 184 152"
          width="184"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <g
              transform="translate(24 31.67)"
            >
              <ellipse
                className="ant-empty-img-default-ellipse"
                cx="67.797"
                cy="106.89"
                rx="67.797"
                ry="12.668"
              />
              <path
                className="ant-empty-img-default-path-1"
                d="M122.034 69.674L98.109 40.229c-1.148-1.386-2.826-2.225-4.593-2.225h-51.44c-1.766 0-3.444.839-4.592 2.225L13.56 69.674v15.383h108.475V69.674z"
              />
              <path
                className="ant-empty-img-default-path-2"
                d="M101.537 86.214L80.63 61.102c-1.001-1.207-2.507-1.867-4.048-1.867H31.724c-1.54 0-3.047.66-4.048 1.867L6.769 86.214v13.792h94.768V86.214z"
                transform="translate(13.56)"
              />
              <path
                className="ant-empty-img-default-path-3"
                d="M33.83 0h67.933a4 4 0 0 1 4 4v93.344a4 4 0 0 1-4 4H33.83a4 4 0 0 1-4-4V4a4 4 0 0 1 4-4z"
              />
              <path
                className="ant-empty-img-default-path-4"
                d="M42.678 9.953h50.237a2 2 0 0 1 2 2V36.91a2 2 0 0 1-2 2H42.678a2 2 0 0 1-2-2V11.953a2 2 0 0 1 2-2zM42.94 49.767h49.713a2.262 2.262 0 1 1 0 4.524H42.94a2.262 2.262 0 0 1 0-4.524zM42.94 61.53h49.713a2.262 2.262 0 1 1 0 4.525H42.94a2.262 2.262 0 0 1 0-4.525zM121.813 105.032c-.775 3.071-3.497 5.36-6.735 5.36H20.515c-3.238 0-5.96-2.29-6.734-5.36a7.309 7.309 0 0 1-.222-1.79V69.675h26.318c2.907 0 5.25 2.448 5.25 5.42v.04c0 2.971 2.37 5.37 5.277 5.37h34.785c2.907 0 5.277-2.421 5.277-5.393V75.1c0-2.972 2.343-5.426 5.25-5.426h26.318v33.569c0 .617-.077 1.216-.221 1.789z"
              />
            </g>
            <path
              className="ant-empty-img-default-path-5"
              d="M149.121 33.292l-6.83 2.65a1 1 0 0 1-1.317-1.23l1.937-6.207c-2.589-2.944-4.109-6.534-4.109-10.408C138.802 8.102 148.92 0 161.402 0 173.881 0 184 8.102 184 18.097c0 9.995-10.118 18.097-22.599 18.097-4.528 0-8.744-1.066-12.28-2.902z"
            />
            <g
              className="ant-empty-img-default-g"
              transform="translate(149.65 15.383)"
            >
              <ellipse
                cx="20.654"
                cy="3.167"
                rx="2.849"
                ry="2.815"
              />
              <path
                d="M5.698 5.63H0L2.898.704zM9.259.704h4.985V5.63H9.259z"
              />
            </g>
          </g>
        </svg>
      </div>
      <p
        className="ant-empty-description"
      >
        No Data
      </p>
    </div>
  </div>
</div>
`;
